include/range/v3/utility/memory.hpp

```diff
@@ -43,11 +43,11 @@ namespace ranges
             {
                 RANGES_EXPECT(count >= 0);
                 std::size_t n = static_cast<std::size_t>(count);
-                while(n > PTRDIFF_MAX / sizeof(T))
-                    n /= 2;
+                if (n > PTRDIFF_MAX / sizeof(T))
+                    n = PTRDIFF_MAX / sizeof(T);
 
                 void *ptr = nullptr;
-                for(; ptr == nullptr && n > 0; n /= 2)
+                for (; ptr == nullptr && n > 0; n /= 2)
```

Comment:
These changes look unrelated.

Reply:
They're related only in that I looked at the code and saw things that need fixing.